### ISSUE TITLE
question about the last epoch's input  batchsize

### ISSUE THREAD

**Issue by yanghhx** [KevinMusgrave/pytorch-metric-learning#379]
My trainset are set to  29346 , testset:  18963 and samplesset:  32. samplers.  the  "m" of MPerClassSampler  are 6,i.e. MPerClassSampler (***, m=6,**))  When running to the last epoch, it comes to the error:the input size:torch.Size([19, *, *, *]). Why my last input size comes to 19? Could you help me?

**Comment by KevinMusgrave** [comment]
Have you set your dataloader's ```drop_last``` to True?

```python
DataLoader(drop_last=True)
```

**Comment by yanghhx** [comment]
Yes. I have set it to True. However, it comes to the confusing result.







At 2021-11-08 02:21:26, "Kevin Musgrave" ***@***.***> wrote:

Have you set your dataloader's drop_last to True?

DataLoader(drop_last=True)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by KevinMusgrave** [comment]
Hmm strange. Can you try it without the sampler and see if you get the same result?

**Comment by yanghhx** [comment]
I will try it.













At 2021-11-09 02:11:57, "Kevin Musgrave" ***@***.***> wrote:

Hmm strange. Can you try it without the sampler and see if you get the same result?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by KevinMusgrave** [comment]
Let me know if you still have issues

**Comment by yanghhx** [comment]
There are no issues for me. Thank you for your answer.


Best Wishes













At 2021-11-16 04:34:27, "Kevin Musgrave" ***@***.***> wrote:

Let me know if you still have issues

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
